src/Motor.Extensions.Hosting.RabbitMQ/BasicPropertiesExtensions.cs

```diff
@@ -1,81 +1,89 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Text;
 using CloudNative.CloudEvents;
-using Motor.Extensions.Hosting.Abstractions;
+using Motor.Extensions.Hosting.CloudEvents;
 using Motor.Extensions.Hosting.RabbitMQ.Options;
 using RabbitMQ.Client;
 
 namespace Motor.Extensions.Hosting.RabbitMQ
 {
     public static class BasicPropertiesExtensions
     {
-        public static readonly string CloudEventPrefix = "cloudEvents:";
+        public static string CloudEventPrefix => "cloudEvents:";
+
+        private static readonly List<CloudEventAttribute> IgnoredAttributes = new();
+
+        static BasicPropertiesExtensions()
+        {
+            IgnoredAttributes.AddRange(RabbitMQBindingExtension.AllAttributes);
+            IgnoredAttributes.AddRange(RabbitMQPriorityExtension.AllAttributes);
+        }
 
         public static void Update<T>(this IBasicProperties self, MotorCloudEvent<byte[]> cloudEvent,
-            RabbitMQPublisherOptions<T> options, ICloudEventFormatter cloudEventFormatter)
+            RabbitMQPublisherOptions<T> options)
         {
-            var messagePriority = cloudEvent.Extension<RabbitMQPriorityExtension>()?.Priority ??
-                                  options.DefaultPriority;
+            var messagePriority = cloudEvent.GetRabbitMQPriority() ?? options.DefaultPriority;
             if (messagePriority.HasValue)
                 self.Priority = messagePriority.Value;
-            var dictionary = new Dictionary<string, object>();
 
-            foreach (var attr in cloudEvent.GetAttributes())
+            var headers = new Dictionary<string, object>();
+
+            foreach (var (key, value) in cloudEvent.GetPopulatedAttributes())
```

Comment:
nit: just as an idea because I don't like `continue`:

```cs
            var attributesToConsider = cloudEvent.GetPopulatedAttributes()
                .Where(t => !IgnoredAttributes.Contains(t.Key));

            foreach (var (key, value) in attributesToConsider)
            {
                headers.Add($"{CloudEventPrefix}{key.Name}", Encoding.UTF8.GetBytes(key.Format(value)));
            }
```